lib/embulk/output/bigquery/value_converter_factory.rb

```diff
@@ -204,12 +204,19 @@ def string_converter
               }
             end
           when 'DATE'
-            Proc.new {|val|
-              next nil if val.nil?
-              with_typecast_error(val) do |val|
-                TimeWithZone.set_zone_offset(Time.parse(val), zone_offset).strftime("%Y-%m-%d")
-              end
-            }
+            if @timestamp_format
+              Proc.new {|val|
+                next nil if val.nil?
+                with_typecast_error(val) do |val|
+                  TimeWithZone.set_zone_offset(Time.parse(val), zone_offset).strftime("%Y-%m-%d")
+                end
+              }
```

Comment:
Using `@timestamp_format` as a flag looks weird because it is not necessary to parse `val` here as TIMESTAMP and DATETIME.

```
Time.strptime(val, @timestamp_format).strftime("%Y-%m-%d %H:%M:%S.%6N")
```